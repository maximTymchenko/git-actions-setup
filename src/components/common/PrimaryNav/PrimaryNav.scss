@import '../../../styles/vars';
@import '../../../styles/mixins';

.primary-nav {
    padding-top: 90px;
    // background: $background__secondary;
    width: 100%;
    //   max-width: $container__size;
    height: 100vh;
    position: fixed;
    top: 65px;
    left: 0;
    z-index: 99;
    display: flex;

    &--filter {
    }

    &__section {
        position: relative;
    }

    &__language {
        position: absolute;
        bottom: 100px;
        right: 50px;
    }

    &__wrapper {
        display: flex;
        justify-content: space-between;
        width: 100%;
        max-width: $container__size;
        margin: 0 auto;
    }

    &__items {
        width: 290px;
        position: relative;

        &--border {
            border-right: 2px solid $secondary;

            @media screen and (max-width: $small__tablet) {
                border: none;
            }
        }
    }

    &__body {
        display: flex;
        align-items: flex-start;
        max-height: 530px;

        @media screen and (max-width: $tablet) {
            max-height: 500px;
            width: 350px;
        }

        @media screen and (max-width: $mobile) {
            border-right: none;
        }
    }
    &__categories {
        display: flex;
        align-items: center;
        cursor: pointer;
        font-family: 'OktaNeue-Black';
        font-size: 28px;
        font-weight: 900;
        color: $secondary;
        margin-bottom: 20px;
        margin-right: 20px;
        border-bottom: 8px solid transparent;

        &:nth-child(6)::after {
            content: 'hiring';
            display: block;
            cursor: pointer;
            text-transform: uppercase;
            font-family: 'OktaNeue-Black';
            margin-top: 5px;
            margin-left: 10px;
            padding: 3px 5px;
            font-size: 10px;
            font-weight: 900;
            line-height: 1.1;
            letter-spacing: 0.5px;
            color: $background__primary;
            background-color: $color__error;
        }

        &:hover {
            color: $primary__light;
        }
        &--active {
            border-bottom: 8px solid $primary__light;
        }

        @media screen and (max-width: $mobile) {
            font-size: 20px;
            width: 180px;
        }
    }

    &__btn--back {
        display: none;

        @media screen and (max-width: $small__tablet) {
            display: block;
            position: relative;
            left: 20px;
        }
    }

    &__subcategories {
        position: relative;
        bottom: 45px;
    }

    &__container {
        border-bottom: 2px solid $secondary;
        margin-bottom: 0px;
        // opacity: 0;
        margin-left: 20px;

        &:first-child {
            @media screen and (max-width: $small__tablet) {
                right: 200px;
            }
        }

        @media screen and (max-width: $tablet) {
            position: absolute;
            z-index: 5;
        }

        @media screen and (max-width: $small__tablet) {
            background: $background__secondary;
            border-left: 2px solid $secondary;
            border-bottom: none;
            right: 160px;
        }

        &--show {
            opacity: 1;
        }
    }
}
