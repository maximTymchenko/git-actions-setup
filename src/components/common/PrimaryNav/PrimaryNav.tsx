import React, { FC, useState, useEffect, useRef } from 'react';
import './PrimaryNav.scss';
import { Link } from 'gatsby';
import Nav from '../Nav';

export interface INavLinks {
    activeLink: null | any;
    links: [
        {
            id: number;
            title: string;
            page?: string | undefined;
        },
    ];
}

interface IPropsPrimaryNav {
    type: string;
    toggleActiveClassArr: any;
    navLinksArray: any;
    toggleNavClass: any;
    toggleActiveClass: any;
    border: string;
    language: string[];
    navLinks: any;
}

const PrimaryNav: FC = (props: IPropsPrimaryNav) => {
    const [navLinks, setNavLinks] = useState<INavLinks>(props.navLinks);

    const [navLinksArray, setNavLinksArray] = useState(props.navLinksArray);

    const toggleNavClassArr = (index: number) => {
        setNavLinksArray({ ...navLinksArray, activeLink: navLinksArray.links[index] });
    };

    const toggleActiveClassArr = (index: number) => {
        if (navLinksArray.links[index] === navLinksArray.activeLink) {
            return 'primary-nav__container primary-nav__container--show';
        } else {
            return 'primary-nav__container';
        }
    };

    const toggleNavClass = (index: number) => {
        setNavLinks({ ...navLinks, activeLink: navLinks.links[index] });
        toggleNavClassArr(index);
    };

    const toggleActiveClass = (index: number) => {
        if (navLinks.links[index] === navLinks.activeLink) {
            return 'primary-nav__categories primary-nav__categories--active';
        } else {
            return 'primary-nav__categories';
        }
    };

    const linkRef = useRef();
    // const [linkState, setLinkState] = useState<boolean>(false);

    // const getElemClass = () => {
    //     const linkClasses = linkRef.current;
    //     // console.log();

    //     if (linkClasses.classList.contains('active__border--style')) {
    //         setLinkState(true);
    //         console.log(linkState);
    //         console.log('YES');
    //     }
    // };

    // console.log(linkRef.current);

    return (
        <div className="primary-nav__section">
            <div data-testid="primary-nav" className={`primary-nav primary-nav--${props.type}`}>
                <ul className="primary-nav__wrapper">
                    <div className="primary-nav__body">
                        <div className={`primary-nav__items primary-nav__items--${props.border}`}>
                            {props.navLinks.links.map((item, index) => (
                                <li
                                    onClick={() => {
                                        toggleNavClass(index);
                                    }}
                                    className={toggleActiveClass(index)}
                                    key={index}
                                >
                                    {item.title}
                                </li>
                            ))}
                        </div>
                        <div className="primary-nav__subcategories">
                            {props.navLinksArray.links.map((array, index, page) => (
                                <div key={index} className={toggleActiveClassArr(index)}>
                                    <Nav
                                        // getElemClass={getElemClass}
                                        // linkRef={linkRef}
                                        typeContainer="primary"
                                        keyValue={array.id}
                                        links={array.array}
                                        page={array.page}
                                    />
                                </div>
                            ))}
                        </div>
                    </div>
                    <div className="primary-nav__language">
                        <Nav links={props.language} />
                    </div>
                </ul>
            </div>
        </div>
    );
};

export default PrimaryNav;
