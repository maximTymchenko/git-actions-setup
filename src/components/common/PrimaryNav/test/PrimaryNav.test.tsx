import React, { useState } from 'react';
import PrimaryNav from '../PrimaryNav';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

const menuLinksServices: INavLinks = {
    activeLink: null,
    links: [
        { id: 1, title: 'Web development' },
        { id: 2, title: 'Cloud services' },
        { id: 3, title: 'Research' },
        { id: 4, title: 'Analytics' },
        { id: 5, title: 'Quality control' },
        { id: 6, title: 'Mobile development' },
    ],
};

const menuLinksIndustries: INavLinks = {
    activeLink: null,
    links: [
        { id: 1, title: 'Retail & E-commerce' },
        { id: 2, title: 'Construction' },
        { id: 3, title: 'Finance' },
        { id: 4, title: 'Media' },
    ],
};

const menuLinksExpertise: INavLinks = {
    activeLink: null,
    links: [
        { id: 1, title: 'SaaS product' },
        { id: 2, title: 'Chatbots' },
        { id: 3, title: 'AR & VR' },
        { id: 4, title: 'Blockchain' },
        { id: 5, title: 'Digital enterprise ' },
        { id: 6, title: 'Drone' },
        { id: 7, title: 'R&D' },
    ],
};

const menuLinksEngagement = {
    activeLink: null,
    links: [
        { id: 1, title: 'On-demand team extension' },
        { id: 2, title: 'Fully managed IT services' },
    ],
};

const navLinks = {
    activeLink: null,
    links: [
        { id: 1, title: 'Services' },
        { id: 2, title: 'Industries' },
        { id: 3, title: 'Expertise' },
        { id: 4, title: 'Engagement' },
    ],
};

const navLinksArray = {
    activeLink: null,
    links: [
        { id: 0, array: menuLinksServices },
        { id: 1, array: menuLinksIndustries },
        { id: 2, array: menuLinksExpertise },
        { id: 3, array: menuLinksEngagement },
    ],
};

const language = {
    activeLink: null,
    links: [
        { id: 1, title: '' },
        { id: 2, title: '' },
    ],
};

it('renders elem correctly', () => {
    const { getByTestId } = render(
        <PrimaryNav navLinks={navLinks} navLinksArray={navLinksArray} language={language} />,
    );
    expect(getByTestId('primary-nav')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer
        .create(
            <PrimaryNav navLinks={navLinks} navLinksArray={navLinksArray} language={language} />,
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
