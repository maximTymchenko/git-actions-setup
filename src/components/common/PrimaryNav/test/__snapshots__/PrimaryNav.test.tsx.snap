// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="primary-nav__section"
>
  <div
    className="primary-nav primary-nav--undefined"
    data-testid="primary-nav"
  >
    <ul
      className="primary-nav__wrapper"
    >
      <div
        className="primary-nav__body"
      >
        <div
          className="primary-nav__items primary-nav__items--undefined"
        >
          <li
            className="primary-nav__categories"
            onClick={[Function]}
          >
            Services
          </li>
          <li
            className="primary-nav__categories"
            onClick={[Function]}
          >
            Industries
          </li>
          <li
            className="primary-nav__categories"
            onClick={[Function]}
          >
            Expertise
          </li>
          <li
            className="primary-nav__categories"
            onClick={[Function]}
          >
            Engagement
          </li>
        </div>
        <div
          className="primary-nav__subcategories"
        >
          <div
            className="primary-nav__btn--back"
          >
            <button
              className="button button--apply--reverse button--undefined button--undefined"
              data-testid="button"
            >
              Back
              <img
                className="button__icon button__icon--undefined"
                data-testid="button__icon"
                src="test-file-stub"
              />
            </button>
          </div>
          <div
            className="primary-nav__container"
          >
            <ul
              className="nav nav--undefined"
            >
              <div
                className="nav__container nav__container--primary"
              >
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Web development
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Cloud services
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Research
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Analytics
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Quality control
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Mobile development
                  </li>
                </a>
              </div>
            </ul>
          </div>
          <div
            className="primary-nav__container"
          >
            <ul
              className="nav nav--undefined"
            >
              <div
                className="nav__container nav__container--primary"
              >
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Retail & E-commerce
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Construction
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Finance
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Media
                  </li>
                </a>
              </div>
            </ul>
          </div>
          <div
            className="primary-nav__container"
          >
            <ul
              className="nav nav--undefined"
            >
              <div
                className="nav__container nav__container--primary"
              >
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    SaaS product
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Chatbots
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    AR & VR
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Blockchain
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Digital enterprise 
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Drone
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    R&D
                  </li>
                </a>
              </div>
            </ul>
          </div>
          <div
            className="primary-nav__container"
          >
            <ul
              className="nav nav--undefined"
            >
              <div
                className="nav__container nav__container--primary"
              >
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    On-demand team extension
                  </li>
                </a>
                <a
                  className="link__margin  active__border nav__link undefined"
                  data-testid="nav"
                  href="/undefined"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <li
                    className="nav__link"
                    onClick={[Function]}
                  >
                    Fully managed IT services
                  </li>
                </a>
              </div>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="primary-nav__language"
      >
        <ul
          className="nav nav--undefined"
        >
          <div
            className="nav__container nav__container--undefined"
          >
            <a
              className="link__margin  active__border nav__link undefined"
              data-testid="nav"
              href="/undefined"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <li
                className="nav__link"
                onClick={[Function]}
              >
                
              </li>
            </a>
            <a
              className="link__margin  active__border nav__link undefined"
              data-testid="nav"
              href="/undefined"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              <li
                className="nav__link"
                onClick={[Function]}
              >
                
              </li>
            </a>
          </div>
        </ul>
      </div>
    </ul>
  </div>
</div>
`;
